tests: kernel: fix unhandled return values.

wentongwu opened the pull request:
Fix coverity issue 203454 and 203507. Fix unhandled return values as most other places of the same function handled in this file.

Fixes: #18443 and #18445.

aescolar commented on tests/kernel/mutex/sys_mutex/src/main.c:
```diff
@@ -429,7 +431,11 @@ void test_main(void)
-	sys_mutex_lock(&not_my_mutex, K_NO_WAIT);
+	rv = sys_mutex_lock(&not_my_mutex, K_NO_WAIT);
+	if (rv != 0) {
+		TC_ERROR("Failed to take mutex %p\n", &not_my_mutex);
```
Is it intended that this will not fail the testcase?
If this ever fails, sanitycheck will not detect this as a failure..
i.e. hacking the condition to `if(1) {` you'd get
```
***** Booting Zephyr OS build zephyr-v1.14.0-3283-g62fa472919a7 *****
FAIL - test_main@436. Failed to take mutex 0x5660f1b4
===================================================================
PROJECT EXECUTION SUCCESSFUL
```

wentongwu replied:
@aescolar Thanks. I will change it.

aescolar commented on tests/kernel/mem_protect/sys_sem/src/main.c:
```diff
@@ -116,8 +116,12 @@ void sem_take_multiple_high_prio_helper(void *p1, void *p2, void *p3)
 
 void sem_multiple_threads_wait_helper(void *p1, void *p2, void *p3)
 {
+	s32_t ret_value;
```
By the way, if you are changing things, maybe it would be better to set this as int to match the sys_sem_take return type

aescolar commented:
@wentongwu Side issue: Could you correct your machine time? The commits time is several hours ahead of reality, which can confuse others quite a bit.

nashif reviewed (changes requested):
@wentongwu please fix your commit messages: "fix coverity issue " is not a proper commit message, please specify what you have fixed in details, nobody cares if this from coverity or not.

nashif commented:
and fix your system time.

wentongwu committed:
tests: kernel: fix unhandled return values.

Fix unhandled return values as most other places handled in this
file, fix coverity issue 203454.

Fixes: #18443.

Signed-off-by: Wentong Wu <wentong.wu@intel.com>

wentongwu committed:
tests: kernel: fix unhandled return values.

Fix unhandled return values as most other places handled in this
file, fix coverity issue 203507.

Fixes: #18445.

Signed-off-by: Wentong Wu <wentong.wu@intel.com>

aescolar reviewed (dismissed):
Seems you forgot the TC_ERROR part? or is that intended?

wentongwu commented:
> Seems you forgot the TC_ERROR part? or is that intended?

that is intended, if lock the mutex failed, unlock will tell the case failed. And this print doesn't change the original execution path. @aescolar

ioannisg commented:
@nashif could you revisit this one?
